addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -91,6 +91,7 @@ def _compute_message_unread(self):
                          ON discuss_channel_member.channel_id = mail_message.res_id
                       WHERE mail_message.model = 'discuss.channel'
                         AND mail_message.message_type NOT IN ('notification', 'user_notification')
+                        AND mail_message.body != ''
```

Comment:
Sadly, this is not enough to detect an empty body.
Look at the `_filter_empty` in mail_message.py.

Reply:
I understand through the existence of the `_filter_empty` method that "deleted" messages with a non empty body exist.
It is also true that the way messages get "deleted" on Discuss in the most recent versions is by setting the body to an empty string.

Since the SQL query in question has the objective of counting unread messages in chat/channels discussions, I believe deletions there only happen by setting the body to an empty string.

Even if for some reasons some messages would still be counted it would still be an edge case and I think an inaccurate counter in rare cases is better then an always inaccurate one.

I also added a check for any attachments.